inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/DynamicSchemaHandleOperator.java

```diff
@@ -314,7 +314,11 @@ private void handleSchemaInfoEvent(TableIdentifier tableId, Schema schema) {
                                 LOG.warn("Ignore table {} schema change, old: {} new: {}.",
                                         tableId, dataSchema, latestSchema, e);
                                 blacklist.add(tableId);
-                                handleDirtyData(jsonNode, jsonNode, DirtyType.EXTRACT_ROWDATA_ERROR, e, tableId);
+                                List<RowData> rowDataForDataSchemaList = dynamicSchemaFormat.extractRowData(jsonNode,
```

Comment:
is there a possibility that this line will throw an exception? If so, is it handled properly?

Reply:
1. dynamicSchemaFormat.extractRowData() usually throws an exception when the schema does not match.
2. Use dataSchema for conversion, usually no exception will be thrown;
3. We can also use try catch to output the error log when an exception is thrown? @Yizhou-Yang